Codigos/Estruturas-de-Dados/XorTrie/XorTrie.cpp

```diff
@@ -0,0 +1,63 @@
+template <int bits = 30> struct XorTrie {
+    vector<vector<int>> go;
+    int root, cnt;
+
+    int new_node() {
+        go[0][cnt] = -1;
+        go[1][cnt] = -1;
+        return cnt++;
+    }
+
+    void insert(int x) {
+        int v = root;
+        for (int i = bits - 1; i >= 0; i--) {
+            if (go[x >> i & 1][v] == -1) {
+                go[x >> i & 1][v] = new_node();
+            }
+            v = go[x >> i & 1][v];
+        }
+    }
+
+#warning se a trie estiver vazia, a query retornara -1
+    int max_xor(int x) {
```

Comment:
```suggestion
    int max_xor(int x) {
         if (cnt <= 1) return -1; // se a trie estiver vazia, retorna -1
```

Reply:
WDYT of combining both functions, the only difference is what is good and what is bad, it could be controlled with a boolean flag and then swapping the values.